itests/eth_config_test.go

```diff
@@ -0,0 +1,64 @@
+// stm: #integration
+package itests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types/ethtypes"
+	"github.com/filecoin-project/lotus/itests/kit"
+)
+
+func TestEthFilterAPIDisabledViaConfig(t *testing.T) {
+	ctx := context.Background()
+
+	kit.QuietMiningLogs()
+
+	// don't pass kit.RealTimeFilterAPI() so ActorEvent.EnableRealTimeFilterAPI is false
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC(), kit.DisableEthRPC())
+	ens.InterconnectAll().BeginMining(10 * time.Millisecond)
+
+	_, err := client.EthNewPendingTransactionFilter(ctx)
+	require.NotNil(t, err)
+	require.Equal(t, err.Error(), api.ErrNotSupported.Error())
```

Comment:
Could register the NotSupported error in https://github.com/filecoin-project/lotus/blob/release/v1.20.0/api/api_errors.go so that it would get passed through jsonrpc with error code, and get unmarshalled into the correct error type.

Entirely optional tho.